From 1afc99d106f705dd0988ce2f18e417b9d653aad0 Mon Sep 17 00:00:00 2001
From: Chris Nauroth <cnauroth@apache.org>
Date: Tue, 10 May 2016 15:14:31 -0700
Subject: [PATCH 1830/2848] HADOOP-12801. Suppress obsolete S3FileSystem
 tests. Contributed by Chris Nauroth.

(cherry picked from commit d3dbfa7feee7922ddcd0b2e97898adf4e0c37de3)
(cherry picked from commit ba4883443141354fa457887fb2501226f6a0fa1f)
(cherry picked from commit bbd463b3b27da44f2e5fea35e04ed71aee763f32)

Change-Id: Id3d5410a1def824b86e5a6747a18aa55f99edc37
---
 .../fs/contract/s3/TestS3ContractRootDir.java      |   15 +++++++++++++++
 .../hadoop/fs/contract/s3/TestS3ContractSeek.java  |    9 +++++++++
 .../org/apache/hadoop/fs/s3/TestS3Credentials.java |    3 +++
 3 files changed, 27 insertions(+)

diff --git a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3/TestS3ContractRootDir.java b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3/TestS3ContractRootDir.java
index b968081..aeb68ba 100644
--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3/TestS3ContractRootDir.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3/TestS3ContractRootDir.java
@@ -22,6 +22,9 @@
 import org.apache.hadoop.fs.contract.AbstractContractRootDirectoryTest;
 import org.apache.hadoop.fs.contract.AbstractFSContract;
 
+import org.junit.Ignore;
+import org.junit.Test;
+
 /**
  * root dir operations against an S3 bucket
  */
@@ -31,4 +34,16 @@
   protected AbstractFSContract createContract(Configuration conf) {
     return new S3Contract(conf);
   }
+
+  @Override
+  @Test
+  @Ignore
+  public void testRmEmptyRootDirNonRecursive() {
+  }
+
+  @Override
+  @Test
+  @Ignore
+  public void testRmRootRecursive() {
+  }
 }
diff --git a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3/TestS3ContractSeek.java b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3/TestS3ContractSeek.java
index bfcd163..5f9535f 100644
--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3/TestS3ContractSeek.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3/TestS3ContractSeek.java
@@ -23,10 +23,19 @@
 import org.apache.hadoop.fs.contract.AbstractFSContract;
 import org.apache.hadoop.fs.contract.ContractTestUtils;
 
+import org.junit.Ignore;
+import org.junit.Test;
+
 public class TestS3ContractSeek extends AbstractContractSeekTest {
 
   @Override
   protected AbstractFSContract createContract(Configuration conf) {
     return new S3Contract(conf);
   }
+
+  @Override
+  @Test
+  @Ignore
+  public void testReadFullyZeroByteFile() {
+  }
 }
diff --git a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3/TestS3Credentials.java b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3/TestS3Credentials.java
index 28e1f4b..bcd52c0 100644
--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3/TestS3Credentials.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3/TestS3Credentials.java
@@ -28,6 +28,7 @@
 import java.net.URI;
 
 import org.junit.Before;
+import org.junit.Ignore;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
@@ -120,6 +121,7 @@ public void testCredentialProvider() throws Exception {
   }
 
   @Test(expected=IllegalArgumentException.class)
+  @Ignore
   public void noSecretShouldThrow() throws Exception {
     S3Credentials s3Credentials = new S3Credentials();
     Configuration conf = new Configuration();
@@ -128,6 +130,7 @@ public void noSecretShouldThrow() throws Exception {
   }
 
   @Test(expected=IllegalArgumentException.class)
+  @Ignore
   public void noAccessIdShouldThrow() throws Exception {
     S3Credentials s3Credentials = new S3Credentials();
     Configuration conf = new Configuration();
-- 
1.7.9.5

